app/lib/annotation_viz_service.rb

```diff
@@ -1,10 +1,46 @@
 class AnnotationVizService
   # set of utility methods used for interacting with annotation data
-  def self.get_selected_annotation(study, cluster, annot_name, annot_type, annot_scope)
+
+  # Retrieves an object representing the selected annotation. If nil is passed for the last four
+  # arguments, it will get the study annotation instead
+  # Params:
+  # - study: the Study object
+  # - cluster: ClusterGroup object (or nil for study-wide annotations)
+  # - annot_name: string name of the annotation
+  # - annot_type: string type (group or numeric)
+  # - annot_scope: string scope (study, cluster, or user)
+  # Returns:
+  # - See populate_annotation_by_class for the object structure
+  def self.get_selected_annotation(study, cluster=nil, annot_name=nil, annot_type=nil, annot_scope=nil)
```

Comment:
I think this refactoring is fine - this method can have a case where the request returns `nil`, either through bad parameters, or the study not having a default annotation.  While this isn't ideal, because `populate_annotation_by_class` is type safe, it can safely return `nil` without throwing an exception.